src/components/document/document.tsx

```diff
@@ -213,8 +168,12 @@ export class DocumentComponent extends BaseComponent<IProps, IState> {
   }
 
   private renderToolbar() {
+    const {document, isGhostUser, readOnly} = this.props;
+    const isPublication = document.isPublished;
+    const showToolbar = this.isPrimary() && !isGhostUser && !readOnly && !isPublication;
```

Comment:
For my edification, what is the definition of a ghost user? Is it a user who can look at documents, but not interact with them (such as a teacher)?

Reply:
Exactly